models/log_test.go

```diff
@@ -177,3 +177,98 @@ func TestPluginOptionDeprecation(t *testing.T) {
 		})
 	}
 }
+
+func TestPluginOptionValueDeprecation(t *testing.T) {
+	info := telegraf.DeprecationInfo{
+		Since:     "1.25.0",
+		RemovalIn: "2.0.0",
+		Notice:    "please check",
+	}
+	var tests = []struct {
+		name     string
+		level    telegraf.Escalation
+		value    interface{}
+		expected string
+	}{
+		{
+			name:     "Error level",
+			level:    telegraf.Error,
+			value:    "foobar",
+			expected: `Value "foobar" for option "option" of plugin "test" deprecated since version 1.25.0 and will be removed in 2.0.0: please check`,
+		},
+		{
+			name:     "Warn level",
+			level:    telegraf.Warn,
+			value:    "foobar",
+			expected: `Value "foobar" for option "option" of plugin "test" deprecated since version 1.25.0 and will be removed in 2.0.0: please check`,
+		},
+		{
+			name:     "nil value",
+			level:    telegraf.Error,
+			value:    nil,
+			expected: `Value "<nil>" for option "option" of plugin "test" deprecated since version 1.25.0 and will be removed in 2.0.0: please check`,
+		},
+		{
+			name:     "Boolean value",
+			level:    telegraf.Error,
+			value:    true,
+			expected: `Value "true" for option "option" of plugin "test" deprecated since version 1.25.0 and will be removed in 2.0.0: please check`,
+		},
+		{
+			name:     "Integer value",
+			level:    telegraf.Error,
+			value:    123,
+			expected: `Value "123" for option "option" of plugin "test" deprecated since version 1.25.0 and will be removed in 2.0.0: please check`,
+		},
+		{
+			name:     "None",
+			level:    telegraf.None,
+			expected: ``,
+		},
+	}
+
+	// Switch the logger to log to a buffer
+	var buf bytes.Buffer
+	scanner := bufio.NewScanner(&buf)
+
+	previous := log.Writer()
+	log.SetOutput(&buf)
+	defer log.SetOutput(previous)
+
+	msg := make(chan string, 1)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			buf.Reset()
+			PrintOptionValueDeprecationNotice(tt.level, "test", "option", tt.value, info)
+			// Wait for a newline to arrive and timeout for cases where
+			// we don't see a message.
+			go func() {
+				scanner.Scan()
+				msg <- scanner.Text()
+			}()
+
+			// Reduce the timeout if we do not expect a message
+			timeout := 1 * time.Second
+			if tt.expected == "" {
+				timeout = 100 * time.Microsecond
+			}
+
+			var actual string
+			select {
+			case actual = <-msg:
+			case <-time.After(timeout):
+			}
+
+			if tt.expected != "" {
+				// Remove the time for comparison
+				parts := strings.SplitN(actual, " ", 3)
+				require.Len(t, parts, 3)
+				actual = parts[2]
+				expected := deprecationPrefix(tt.level) + ": " + tt.expected
+				require.Equal(t, expected, actual)
+			} else {
+				require.Empty(t, actual)
+			}
```

Comment:
This is the code using `require.Eventually`. If possible please also replace the other instances to serve as better examples. ;-)

```suggestion
			// Wait for the message to arrive
			if tt.expected == "" {
				// For empty messages we need to wait the maximum time to
				// make sure nothing arrived.
				time.Sleep(timeout)
				require.Empty(t, buf.String())
				return
			}
			require.Eventually(t, func() bool {
				return strings.HasSuffix(buf.String(), "\n")
			}, timeout, 100*time.Millisecond)

			// Check the result and remove the newline and time for comparison
			parts := strings.SplitN(strings.TrimSpace(buf.String()), " ", 3)
			require.Len(t, parts, 3)
			actual := parts[2]
			expected := deprecationPrefix(tt.level) + ": " + tt.expected
			require.Equal(t, expected, actual)
```

Reply:
Thanks, done.